week1/homework/src/server.js

```diff
@@ -8,8 +8,33 @@ const http = require('http');
 function createServer(port) {
   let state = 10;
 
+  function changeState(response) {
+    response.writeHead(200, { 'Content-Type': 'application/json' });
+    response.end(JSON.stringify({ state }));
+  }
+
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    switch (request.url) {
+      case '/state':
+        changeState(response);
+        break;
+      case '/add':
+        state += 1;
```

Comment:
nice that you kept the business logic (i.e. increment) separate from the technical details such as http status and headers